tests/rest/client/test_rooms.py

```diff
@@ -701,6 +704,30 @@ async def user_may_join_room(
 
         self.assertEqual(join_mock.call_count, 0)
 
+    def test_spam_checker_may_join_room(self) -> None:
+        """Tests that the user_may_join_room spam checker callback is correctly bypassed
+        when creating a new room.
+        """
+
+        async def user_may_join_room(
+            mxid: str,
+            room_id: str,
+            is_invite: bool,
+        ) -> Codes:
+            return Codes.CONSENT_NOT_GIVEN
+
+        join_mock = Mock(side_effect=user_may_join_room)
```

Comment:
Note that this can be inlined this way:

```python
join_mock = Mock(return_value=make_awaitable(Codes.CONSENT_NOT_GIVEN))
```

Reply:
I'm copying and pasting existing code, didn't want to add new subtleties :)